### PR TITLE
RFC: Modified test scripts to support both Python2 and Python3

### PR DESCRIPTION
**Pull request by nasastry**
Modified test scripts to support both Python2 and Python3.
Not Tested.
Followed official porting guide:
https://docs.python.org/3/howto/pyporting.html

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
@clebergnu @PraveenPenguin @narasimhan-v Here is the RFC, to support both Python2 and Python3. Not yet tested.

**Comment by nasastry** [comment]
There are some other changes like adding the following lines by default to all the `*.py` files, which I felt unnecessary at this moment. I have not included the following changes in this RFC patch.

```
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from builtins import open
+from future import standard_library
+standard_library.install_aliases()
```

**Comment by nasastry** [comment]
Travis failed with the following line, which can be fixed easily.
`cpu/dwh.py:46:        self.iothreads = self.params.get('iothrd', default=old_div(self.minthreads,2))`

File: cpu/dwh.py [hunk 1]
```diff
@@ -41,9 +43,9 @@ def setUp(self):
         self.minthreads = self.params.get(
             'minthrd', default=(500 + cpu.online_cpus_count()))
         self.maxthreads = self.params.get('maxthrd', default=None)
-        self.iothreads = self.params.get('iothrd', default=self.minthreads/2)
+        self.iothreads = self.params.get('iothrd', default=old_div(self.minthreads,2))
```
Comment (harish-24) [inline cpu/dwh.py]: Please provide a space here `old_div(self.minthreads, 2)`

File: generic/interbench.py [hunk 2]
```diff
@@ -49,8 +51,8 @@ def setUp(self):
                     sm_manager.install(pkg)):
                 self.cancel("%s is needed for the test to be run" % pkg)
 
-        disk_free_mb = (disk.freespace(self.teststmpdir) / 1024) / 1024
-        if memory.memtotal()/1024 > disk_free_mb:
+        disk_free_mb = old_div((old_div(disk.freespace(self.teststmpdir), 1024)), 1024)
+        if old_div(memory.memtotal(),1024) > disk_free_mb:
```
Comment (harish-24) [inline generic/interbench.py]: Here too, It would make travis build successful.

File: cpu/dwh.py (hunk 1 above)
Comment (nasastry) [reply cpu/dwh.py]: @harish-24 noticed it, will take care in the main patch. In this RFC I would be expecting comments on the approach.

File: generic/interbench.py (hunk 2 above)
Comment (nasastry) [reply generic/interbench.py]: @harish-24 noticed it, will take care in the main patch. In this RFC I would be expecting comments on the approach.

**Comment by nasastry** [comment]
@PraveenPenguin removed un-necessary builtin imports. Please have a look and review.

**Comment by PraveenPenguin** [comment]
@nasastry sorry for late will review and post review comment

**Comment by PraveenPenguin** [comment]
@nasastry one more help if it sounds good to you can you please send test case PR categories wise

**Comment by PraveenPenguin** [comment]
closing this PR as now it not relevant as we are moving all test cases python3 based

**Comment by PraveenPenguin** [comment]
@nasastry thanks for your valuable input as RFC

**Pull request closed without merging**